examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -1180,6 +1180,55 @@ void delete_examples(mongocxx::database db) {
     }
 }
 
+void snapshot_examples(mongocxx::client& client) {
+    // Start Example 59
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+    using mongocxx::v_noabi::pipeline;
+
+    auto db = client["pets"];
+
+    // seed 'pets' database with dogs and cats
+    db.drop();
+    db["cats"].insert_one(make_document(kvp("adoptable", true)));
+    db["dogs"].insert_one(make_document(kvp("adoptable", true)));
+    db["dogs"].insert_one(make_document(kvp("adoptable", false)));
+
+    int64_t adoptable_pets_count = 0;
+
+    auto opts = mongocxx::options::client_session{};
+    opts.snapshot(true);
+    auto session = client.start_session(opts);
+
+    {
+        pipeline p;
+
+        p.match(make_document(kvp("adoptable", true))).count("adoptableCatsCount");
+        auto cursor = db["cats"].aggregate(session, p);
+
+        for (auto doc : cursor) {
+            adoptable_pets_count += doc.find("adoptableCatsCount")->get_int32();
```

Comment:
Is there a clean one liner to get either int32 or int64 and return that value as int64 (similar to bson_iter_as_int64 from the bson C API?)

Reply:
AFAIK no.